maint/local_python/binding_c.py

```diff
@@ -887,8 +883,10 @@ def dump_function(func, kind):
         func['_impl_arg_list'].append(RE.m.group(1))
         func['_impl_param_list'].append(extra_param)
     else:
-        s = re.sub(r'MPI(X?)_([a-zA-Z0-9_]*\()', r'internal\1_\2', s, 1)
-        s = "static " + s
+        # The static internal function should not be in the PMPI library
+        G.out.append("#endif /* MPICH_MPI_FROM_PMPI */")
```

Comment:
I like the comment. But this code should be in the same function that dumps `#ifndef MPICH_MPI_FROM_PMPI `. It may mean that you have to add this `if kind=="polymorph"` branching logic again in the upper function, but I think it is worthwhile to avoid splitting the logic of `MPICH_MPI_FROM_PMPI` far apart.

Reply:
I agree that this was kinda gross. I think the new version does a better job of keeping the entries and exit paired (one set in the profiling function and another in the main function).